xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -272,7 +274,16 @@ void CDirectoryProvider::OnEvent(const ADDON::AddonEvent& event)
         typeid(event) == typeid(ADDON::AddonEvents::Disabled) ||
         typeid(event) == typeid(ADDON::AddonEvents::InstalledChanged) ||
         typeid(event) == typeid(ADDON::AddonEvents::MetadataChanged))
-    m_updateState = INVALIDATED;
+      m_updateState = INVALIDATED;
+  }
+}
+
+void CDirectoryProvider::Notify(const Observable &obs, const ObservableMessage msg)
+{
+  if (URIUtils::IsProtocol(m_currentUrl, "pvr"))
+  {
+    if (msg == ObservableMessageManagerStateChanged && g_PVRManager.IsStarted())
```

Comment:
I don't get your point. What exactly is *wrong* with calling g_PVRManager? Need to know this object and thus pvr details anyway in order to register as observer at the pvr manager singleton. Maybe it is not the nicest api, but this is how pvr manager works today - sending one state changed event and offering several methods to obtain the current state. I would like to keep this code as it is now and put pvr messaging rework on my list for v18.